src/diffusers/configuration_utils.py

```diff
@@ -104,9 +136,98 @@ def save_config(self, save_directory: Union[str, os.PathLike], push_to_hub: bool
         logger.info(f"Configuration saved in {output_config_file}")
 
     @classmethod
-    def from_config(cls, pretrained_model_name_or_path: Union[str, os.PathLike], return_unused_kwargs=False, **kwargs):
+    def from_config(cls, config: Union[FrozenDict, Dict[str, Any]] = None, return_unused_kwargs=False, **kwargs):
         r"""
-        Instantiate a Python class from a pre-defined JSON-file.
+        Instantiate a Python class from a config dictionary
+
+        Parameters:
+            config (`Dict[str, Any]`):
```

Comment:
No `FrozenDict` here, see above.

Reply:
Yeah, leaving as is for now since I don't think users need to understand the concept just yet